core/deployment/src/main/java/io/quarkus/deployment/util/ArtifactResourceResolver.java

```diff
@@ -0,0 +1,84 @@
+import java.util.List;
+
+import io.quarkus.maven.dependency.ResolvedDependency;
+import io.quarkus.paths.PathTree;
+
+/**
+ * Utility class to extract a list of resource paths from a given artifact and folder.
+ */
+public final class ArtifactResourceResolver {
+    private final PathTree pathTree;
+
+    /**
+     *
+     * Creates a {@code ArtifactResourceResolver} for the given artifact.
+     *
+     * @param dependencies the resolved dependencies for the build.
+     * @param artifactCoordinates the coordinates of the dependency to resolve.
+     */
+    public static ArtifactResourceResolver of(
+            Collection<ResolvedDependency> dependencies, ArtifactCoordinates artifactCoordinates) {
+
+        return new ArtifactResourceResolver(dependencies, artifactCoordinates);
+    }
+
+    private ArtifactResourceResolver(
+            Collection<ResolvedDependency> dependencies, ArtifactCoordinates artifactCoordinates) {
+
+        var artifact = dependencies.stream()
+                .filter(d -> artifactCoordinates.equals(new ArtifactCoordinates(d.getGroupId(), d.getArtifactId())))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("%s:%s artifact not found"
+                        .formatted(artifactCoordinates.groupId, artifactCoordinates.artifactId)));
+        pathTree = artifact.getContentTree();
+    }
+
+    /**
+     * Extracts a list of resource paths at the given location.
+     *
+     * @param resourceCoordinates the path and extension of the resources to find
+     * @return a stream of the found resource paths
+     */
+    public List<Path> resourcePathList(ResourceCoordinates resourceCoordinates) {
+        return pathTree.apply(resourceCoordinates.basePath(), visit -> {
+            try (var pathStream = Files.list(visit.getPath())) {
+                return pathStream
+                        .map(p -> p.subpath(0, p.getNameCount()))
+                        .filter(p -> p.toString().endsWith(resourceCoordinates.extension()))
+                        .toList();
+            } catch (IOException ex) {
+                throw new RuntimeException(ex);
+            }
+        });
+    }
+
+    /**
+     * Extracts a list of resource paths as strings at the given location
+     *
+     * @param resourceCoordinates the path and extension of the resources to find
+     * @return a list of the found resource paths as strings
+     */
+    public List<String> resourceList(ResourceCoordinates resourceCoordinates) {
+        return resourcePathList(resourceCoordinates).stream()
+                .map(Path::toString)
+                .toList();
+    }
+
+    public record ArtifactCoordinates(String groupId, String artifactId) {
+        public static ArtifactCoordinates of(String groupId, String artifactId) {
+            return new ArtifactCoordinates(groupId, artifactId);
+        }
+    }
+
+    public record ResourceCoordinates(String basePath, String extension) {
```

Comment:
This also looks questionable to me. It seems to be an attempt to introduce a generic way of locating content but the way it works remains pretty specific. Have you considered https://github.com/quarkusio/quarkus/blob/main/independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/PathFilter.java?

Reply:
Thanks for the pointer. Done.